### DISCUSSION TITLE
Can I connect to the database (Mysql) through the socks proxy?

### DISCUSSION THREAD

**Discussion by hewx815** [prisma/prisma#23572] [Q&A] [answered]
### Question

Can I connect to the database (Mysql) through the socks proxy?

This is some of my previous code:
```js
import mysql2 from 'mysql2';
import type { Connection } from 'mysql2/promise';
import SocksConnection from 'socksjs';

function createSocksConnection(options: SocksConnectionOptions) {
  const socks = new SocksConnection({
    host: options.sqlHost,
    port: options.sqlPort,
  }, {
    host: options.socksHost,
    port: options.socksPort,
  });

  const connection = mysql2.createConnection({
    user: options.sqlUsername,
    password: options.sqlPassword,
    database: options.sqlDatabase,
    stream: socks,
  });

  return connection.promise();
}
```

**Comment by ludralph** [comment, marked as answer]
Hi @hewx815 👋 

Thank you for raising this question.

Prisma does not support connecting to a MySQL database through a SOCKS proxy. However, if you want to see this supported, you can create a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) and document your usecase so our engineers can take a look.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
